python/cudf/cudf/core/series.py

```diff
@@ -1814,20 +1814,14 @@ def fillna(
     ):
         if isinstance(value, pd.Series):
             value = Series.from_pandas(value)
-
-        if not (is_scalar(value) or isinstance(value, (abc.Mapping, Series))):
-            raise TypeError(
-                f'"value" parameter must be a scalar, dict '
-                f"or Series, but you passed a "
-                f'"{type(value).__name__}"'
-            )
-
-        if isinstance(value, (abc.Mapping, Series)):
+        elif isinstance(value, abc.Mapping):
             value = Series(value)
+        if isinstance(value, cudf.Series):
             if not self.index.equals(value.index):
                 value = value.reindex(self.index)
-            value = value._column
-
+            value = {
+                col_name: value._column for col_name in self._column_names
+            }
```

Comment:
*question*: Does this do the right thing if `self._column_names` has more than one entry? Should it not be:

```
(name, ) = self._data.keys()
value = {name: value._column}
```
?

Reply:
I think `Series._column_names` _should_ only every have 1 entry, but I'll switch to your suggestion in order to be safe